sysinfo.ts

```diff
@@ -0,0 +1,99 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import os = require("os");
+import osenv = require("osenv");
+import Future = require("fibers/future");
+import hostInfo = require("./host-info");
+
+export class SysInfo implements ISysInfo {
+	constructor(private $childProcess: IChildProcess) { }
+
+	private sysInfoCache: ISysInfoData = undefined;
+
+	getSysInfo(): ISysInfoData {
+		if (!this.sysInfoCache) {
+			var res: ISysInfoData = Object.create(null);
+
+			var packageJson = require("../../package.json");
+			res.procInfo = packageJson.name + " / " + packageJson.version;
+
+			// os stuff
+			res.platform = os.platform();
+			res.os = hostInfo.isWindows() ? this.winVer() : this.unixVer();
+			res.shell = osenv.shell();
+			res.dotNetVer = hostInfo.isWindows() ? hostInfo.dotNetVersion(".Net is not installed").wait() : undefined;
```

Comment:
Is it better to move the `isWindows()` in the hostInfo implementation? In this case if we need this information on another place we will not have to do the ternary again.